rflx/generator/generator.py

```diff
@@ -1939,49 +1967,58 @@ def __create_reset_dependent_fields_procedure(message: Message) -> UnitPart:
                         ),
                     ],
                     [
-                        field_location_invariant,
-                        CaseStatement(
-                            Variable("Fld"),
-                            [
-                                (
-                                    Variable(f.affixed_name),
-                                    ty.cast(ty.List[Statement], [])
-                                    + [
+                        PragmaStatement("Assert", [field_location_invariant]),
+                        *(
+                            []
+                            if len(message.fields) == 1
+                            else [
+                                ForIn(
+                                    "F_Loop",
+                                    ValueRange(Succ("Field", Variable("Fld")), Last("Field")),
+                                    [
                                         Assignment(
                                             Indexed(
                                                 Variable("Ctx.Cursors"),
-                                                Variable(s.affixed_name),
+                                                Variable("F_Loop"),
```

Comment:
The name `F_Loop` will lead to a name conflict, if a message field is called `Loop`. The prefix `F_` must be reserved for the literals in the `Virtual_Field` type to prevent name conflicts.

Reply:
I use "Fld_Loop" in c9bccc8